morpheus/models/dfencoder/dataloader.py

```diff
@@ -352,89 +349,46 @@ def get_preloaded_data(self):
             self._preloaded_data = {fn: self._load_data_fn(f"{self._data_folder}/{fn}") for fn in self._filenames}
         return pd.concat(pdf for pdf in self._preloaded_data.values())
 
-    @staticmethod
-    def get_train_dataset(model, data_folder, load_data_fn=pd.read_csv, preload_data_into_memory=False):
-        """A helper function to get a train dataset with the provided parameters.
+    @property
+    def preprocess_fn(self):
+        return self._preprocess_fn
 
-        Parameters
-        ----------
-        model : AutoEncoder
-            The autoencoder model used to get relevant params and the preprocessing func.
-        data_folder : str
-            The path to the folder containing the data.
-        load_data_fn : function, optional
-            A function for loading data from a provided file path into a pandas.DataFrame, by default pd.read_csv.
-        preload_data_into_memory : bool, optional
-            Whether to preload all the data into memory, by default False.
+    @preprocess_fn.setter
+    def preprocess_fn(self, value):
+        self._preprocess_fn = value
 
-        Returns
-        -------
-        DatasetFromPath
-            Validation Dataset set up to load from the path.
-        """
-        dataset = DatasetFromPath(
-            data_folder,
-            model.batch_size,
-            model.preprocess_train_data,
-            load_data_fn=load_data_fn,
-            shuffle_rows_in_batch=True,
-            shuffle_batch_indices=True,
-            preload_data_into_memory=preload_data_into_memory,
-        )
-        return dataset
+    @property
+    def batch_size(self):
+        return self._batch_size
 
-    @staticmethod
-    def get_validation_dataset(model, data_folder, load_data_fn=pd.read_csv, preload_data_into_memory=True):
-        """A helper function to get a validation dataset with the provided parameters.
+    @batch_size.setter
+    def batch_size(self, value):
+        self._batch_size = value
```

Comment:
I feel like you shouldnt be able to change the batch size after the dataset has been created? If its in the middle of creating batches, this could cause a problem, right?

Reply:
Yes, although we don't make any kind of contract that this is thread safe, so if someone is changing batch size while iterating that seems like an algorithmic problem; we also do change the batch size and various other config params when we transition from doing training to the validation step.